tests/LargeXlsx.Tests/XlsxWriterTest.cs

```diff
@@ -447,4 +447,41 @@ public static void SharedStrings()
             sheet.Cells["A6"].Value.Should().Be("Lorem ipsum dolor sit amet");
         }
     }
+
+    [Test]
+    public static void HeaderFooter()
+    {
+        using var stream = new MemoryStream();
+        using (var xlsxWriter = new XlsxWriter(stream))
+        {
+            var headerFooter = 
+                new XlsxHeaderFooter(
+                    alignWithMargins: true,
+                    differentFirst: true,
```

Comment:
Rather than having a boolean for differentFirst and differentOddEven, could we deduce it from other fields?
For example, I'm thinking to an optional First property that, when used, would imply differentFirst. Same for, let's say, Even. That would reduce the number of invalid representable states, hence the need to check for them, and hopefully make the API clearer.

Reply:
I got rid of the `differentFirst` and `differentOddEven` bools. `differentFirst` is now set true if `FirstHeader` or `FirstFooter` have been set. Same with `differentOddEven`, if `EvenHeader` or `EvenFooter` is used the property will be set true.

Please take a look what you think about the implementation or maybe you'l see any clear problems with it then let me know.